src/test/java/BaseTest.java

```diff
@@ -1,18 +1,142 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.apache.commons.lang3.ObjectUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
-import org.openqa.selenium.firefox.FirefoxDriver;
-import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.interactions.Action;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+
+import java.time.Duration;
 
 
 public class BaseTest {
+   public static WebDriver driver = null;
+   public static String url = null;
+   public static WebDriverWait wait = null;
 
+   public static Actions actions = null;
+  
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    @BeforeMethod
+    @Parameters({"baseUrl"})
+    public void launchBrowser(String baseUrl){
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--disable-notifications","--remote-allow-origins=*","--incognito","--start-maximize");
+
+        driver = new ChromeDriver(options);
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+        url = baseUrl;
+        driver.get(url);
+        
+    }
+
+    public static void provideEmail(String email){
+     WebElement emailField = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='email']")));
+     emailField.click();
+     emailField.sendKeys(email);
+     
+    }
+      public static void providePassword(String password){
+        WebElement passwordField = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='password']")));
+        passwordField.click();
+        passwordField.sendKeys(password);
+      }
+    
+      public static void clickSubmit(){
+        WebElement submitButton = wait.until(ExpectedConditions.elementToBeClickable
+                (By.cssSelector("[type='submit']")));
+        submitButton.click();
+  
+      }
+      public static void logIn(String email, String password){
+        provideEmail(email);
+        providePassword(password);
+        clickSubmit();
+        
+      }
```

Comment:
BaseTest — Implements the necessary methods for work with webdriver.

We will store the web elements present on each page in the corresponding java class.

Please, move it and another methods in pages



![image](https://user-images.githubusercontent.com/104761482/226459999-1952217b-72a6-4ff8-aa2e-1ff1af20274f.png)